src/test/java/database/service/JsonDatabaseServiceTest.java

```diff
@@ -115,6 +115,22 @@ public void removeAllRecordsFromTableTest() {
         assertTrue(jsonDatabaseService.deleteTable(Student.class));
     }
 
+    @Test
+    public void removeAllRecordsResetIdCounterCheckTest() {
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+        jsonDatabaseService.addNewRecordToTable(secondStudent);
+
+        jsonDatabaseService.removeAllRecordsFromTable(Student.class);
+        Path databasePath = Path.of(jsonDatabaseService.getDatabasePath(Student.class));
+        assertEquals(EMPTY_BRACKETS_TO_JSON, jsonDatabaseService.readDatabaseFile(databasePath));
+
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+        assertEquals(firstStudent, jsonDatabaseService.getById(Student.class, 0));
```

Comment:
Please add another table in this test to check that id counter is reset only for requested table.